flow/project.py

```diff
@@ -261,18 +261,14 @@ class JobOperation(object):
         :class:`dict`
     """
 
-    def __init__(self, id, name, job, cmd, directives=None):
+    def __init__(self, id, name, job, cmd, directives):
         self._id = id
         self.name = name
         self.job = job
         if not (callable(cmd) or isinstance(cmd, str)):
             raise ValueError("JobOperation cmd must be a callable or string.")
         self._cmd = cmd
-
-        if directives is None:
```

Comment:
From previous comment, this logic for checking if directives have been set is still needed.